packages/addresses/src/curve.ts

```diff
@@ -0,0 +1,24 @@
+import { pointIsOnCurve } from './vendor/noble/ed25519';
+
+function byteToHex(byte: number): string {
+    const hexString = byte.toString(16);
+    if (hexString.length === 1) {
+        return `0${hexString}`;
+    } else {
+        return hexString;
+    }
+}
+
+function decompressPointBytes(bytes: Uint8Array): bigint {
+    const hexString = bytes.reduce((acc, byte, ii) => `${byteToHex(ii === 31 ? byte & ~0x80 : byte)}${acc}`, '');
+    const integerLiteralString = `0x${hexString}`;
+    return BigInt(integerLiteralString);
+}
+
+export async function compressedPointBytesAreOnCurve(bytes: Uint8Array): Promise<boolean> {
+    if (bytes.byteLength !== 32) {
+        return false;
+    }
+    const y = decompressPointBytes(bytes);
+    return pointIsOnCurve(y, bytes[31]);
+}
```

Comment:
Is there any explanation for the code in this file we could add a link to as a comment? Are these standard/well known ECC algorithms for example?

Reply:
Yeah, I actually put the comment to the algorithm above the implementation of `pointIsOnCurve()` in `vendor/noble/ed25519.ts`

```ts
https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.3
export function pointIsOnCurve(y: bigint, lastByte: number): boolean {
```